### PR TITLE
Issue #295 - validate existing files using checksum in files.download

### PR DESCRIPTION
**Pull request by sysadmin75**
Added functionality for sha256sum and md5sum for files.download.

### CODE REVIEW

**Comment by sysadmin75** [comment]
I'm not sure how to resolve the test failure.  I'm open to suggestions.

**Comment by sysadmin75** [comment]
I was able to resolve the test locally by reverting this commit:
https://github.com/Fizzadar/pyinfra/pull/405/commits/f028fddf7363947ba1fa37f62acbfb9ce4a7a23b
However this will cause the flake8 failure.

**Comment by sysadmin75** [comment]
This change allows both the flake8 and pytest to run.  However this seems like an added step that shouldn't be needed.
```
--- a/pyinfra/facts/files.py
+++ b/pyinfra/facts/files.py
@@ -157,8 +157,8 @@ class Sha256File(FactBase):
     def command(self, name):
         name = escape_unix_path(name)
         self.name = name
-        return 'sha256sum {0} 2> /dev/null || shasum -a 256 {0} 2> /dev/null \
-                || sha256 {0}'.format(name)
+        ret_str = 'sha256sum {0} 2> /dev/null || shasum -a 256 {0} 2> /dev/null || sha256 {0}'
+        return ret_str.format(name)
 
     def process(self, output):
         for regex in self._regexes:
```

File: pyinfra/facts/files.py [hunk 1]
```diff
@@ -138,7 +138,57 @@ def process(self, output):
...
+    _regexes = [
+        r'^([a-zA-Z0-9]{64})\s+%s$',
+        r'^SHA256\s+\(%s\)\s+=\s+([a-zA-Z0-9]{64})$',
+    ]
+
+    def command(self, name):
+        name = escape_unix_path(name)
+        self.name = name
+        return 'sha256sum {0} 2> /dev/null || shasum -a 256 {0} 2> /dev/null \
+                || sha256 {0}'.format(name)
```
Comment (Fizzadar) [inline pyinfra/facts/files.py:161]: This should cover both flake8 + test:

```suggestion
        return (
            'sha256sum {0} 2> /dev/null '
            '|| shasum -a 256 {0} 2> /dev/null '
            '|| sha256 {0}'
         ).format(name)
```

**Pull request merged**